src/js/models/data_filter_model.js

```
@@ -0,0 +1,120 @@
+        this._selectedSubindicators = {}
+        this._childData = childData;
+        this._filteredData = {};
+
+        groups.forEach(group => {
+            let dataFilter = new DataFilter(group);
+            self.filters.push(dataFilter);
+
+            this.groupLookup[dataFilter.name] = dataFilter;
+        });
+    }
+
+    get aggregatableGroups() {
+        return Object.values(this.groupLookup).filter(group => group.can_aggregate);
+    }
+
+    get nonAggregatableGroups() {
+        return Object.values(this.groupLookup).filter(group => !(group.can_aggregate));
+    }
+
+    get selectedFilters() {
+        return this._selectedFilters;
+    }
+
+    get primaryGroup() {
+        return this._primaryGroup;
+    }
+
+    get selectedSubIndicators() {
+        return this._selectedSubindicators;
+    }
+
+    get childData(){
+        return this._childData;
+    }
+
+    get filteredData(){
+        return this._filteredData;
+    }
+
+    get availableFilters() {
+        const self = this;
+        let filters = Object.values(this.groupLookup);
+
+        return filters.filter(filter => {
+            if (filter.name == this.primaryGroup)
+                return false;
+
+            return !(self.selectedFilters.has(filter));
+        });
+    }
+
+    get availableFilterNames() {
+        let filters = [...this.availableFilters]
+        return filters.map(f => f.name)
+    }
+
+    addFilter(indicatorName) {
+        let dataFilter = this.groupLookup[indicatorName];
+        if (dataFilter != undefined) {
+            this._selectedFilters.add(dataFilter);
+            this.triggerEvent(DataFilterModel.EVENTS.updated, this)
+        } else {
+            throw `addFilter: Can't find indicator: ${indicatorName}`
+        }
+    }
+
+    removeFilter(indicatorName) {
+        let dataFilter = this.groupLookup[indicatorName];
+        if (dataFilter != undefined) {
+            this._selectedFilters.delete(dataFilter)
```

should we throw an error if that indicator isn't selected?